### PR TITLE
Create backend classes for express checkout element

### PR DESCRIPTION
**Pull request by Mayisha**
This PR prepares the backend for #3404 and #3403 

## Changes proposed in this Pull Request:
- Created the PHP classes for the express checkout element.
    - `WC_Stripe_Express_Checkout_Element`
    - `WC_Stripe_Express_Checkout_Ajax_Handler`
    - `WC_Stripe_Express_Checkout_Helper`
- All the functions in the above classes are copied from `WC_Stripe_Payment_Request` [class](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-payment-request.php), just broken down to three separate classes.
    - Discarded a couple of unnecessary checks ([example](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-payment-request.php#L1781)).
    - Discarded the [branded button](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-payment-request.php#L295) and [custom button](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/payment-methods/class-wc-stripe-payment-request.php#L317) related functions which we no longer offer in the settings page.

## Testing instructions
- Run `npm install`, `npm start`.
- Enable Google/Apple Pay from the Stripe settings page.
- As a shopper, add a product to your cart and go to the checkout page. Confirm that Google/Apple Pay button loads on both shortcode and block checkout the pages.

Screenshot 2024-09-11 at 2 32 05 PM

- Check the JS scripts loaded in the network tab and confirm that `assets/js/stripe-payment-request.js` is loaded and `build/express_checkout.js` is not loaded.
- Now enable the ECE feature flag (return true from `is_stripe_ece_enabled` function or in your database add an option `_wcstripe_feature_ece` and set `yes` as the value).
- Refresh the checkout page. Confirm that `build/express_checkout.js` script is loaded but `assets/js/stripe-payment-request.js` is not loaded.
- Confirm that Google/Apple Pay button loads on the block checkout page only.

Screenshot 2024-09-11 at 2 30 00 PM

### CODE REVIEW

File: woocommerce-gateway-stripe.php [hunk 1]
```diff
@@ -250,10 +260,16 @@ public function init() {
...
-				$this->account                       = new WC_Stripe_Account( $this->connect, 'WC_Stripe_API' );
+				$this->api                            = new WC_Stripe_Connect_API();
+				$this->connect                        = new WC_Stripe_Connect( $this->api );
+				$this->payment_request_configuration  = new WC_Stripe_Payment_Request();
+				$this->account                        = new WC_Stripe_Account( $this->connect, 'WC_Stripe_API' );
+
+				// Express checkout configurations.
+				$express_checkout_helper = new WC_Stripe_Express_Checkout_Helper();
+				$express_checkout_ajax_handler = new WC_Stripe_Express_Checkout_Ajax_Handler( $express_checkout_helper );
+				$this->express_checkout_configuration = new WC_Stripe_Express_Checkout_Element( $express_checkout_ajax_handler, $express_checkout_helper );
```
Comment (james-allan) [inline woocommerce-gateway-stripe.php:271]: There are some minor alignment issues here and the block above. They all might be aligned with the `$this->express_checkout_configuration = ...`

But the whole block should be together or just aligned with their block.

**Review by james-allan — APPROVED** [review]
I followed the steps to test in the PR description and confirmed that with the ece feature flag disabled/enabled, the relevant scripts are loading and the old ones aren't. 

With the ECE feature flag enabled I wasn't able to confirm the Google/Apple pay button shows on checkout. I'm not sure what's going on with my test site but I was able to get Apple pay working but not google pay.

Just noting that i did notice the following console log error:


```
woocommerce-gateway-stripe/build/express_checkout.css?ver=a027ce0f2b4dbad050b53920624f96f5 404 (Not Found)
```

This is caused by the empty entrypoints/express-checkout/index.js file with no relevant css. Seems expected for now but wanted to flag it. 

This PR is super long so I haven't reviewed each line but given it a cursory look over.

File: woocommerce-gateway-stripe.php (hunk 1 above)
Comment (Mayisha) [reply woocommerce-gateway-stripe.php:271]: Fixed it in the other PR [4dcfe4a](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3439/commits/4dcfe4a39b36e53b4dca89c99304f43285bcf1e0)

**Comment by Mayisha** [comment]
> With the ECE feature flag enabled I wasn't able to confirm the Google/Apple pay button shows on checkout. I'm not sure what's going on with my test site but I was able to get Apple pay working but not google pay.

We need to register the ECE buttons separately. In this PR I have registered the button once only, which is probably why you see only one button.

**Pull request merged**